packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -154,6 +166,16 @@ export class EngineReportingExtension<TContext = any>
       });
     }
 
+    // While clientAddress could be a part of the protobuf, we'll ignore it for
+    // now, since the backend does not group by it and Engine frontend will not
+    // support it in the short term
+    const { clientName, clientVersion } = this.generateClientInfo({
+      context: o.context,
+      extensions: o.extensions,
+    });
+    this.trace.clientName = clientName || '';
```

Comment:
Should we do empty string for unrecognized clients or have a designated enum, like "UNKNOWN"? Either way, we should think about what happens with clients that specify their client information as the default we use for unknown clients -- is that OK?